pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -610,7 +611,16 @@ public RoutingTable getRoutingTable(BrokerRequest brokerRequest, long requestId)
       return null;
     }
     InstanceSelector.SelectionResult selectionResult = routingEntry.calculateRouting(brokerRequest, requestId);
-    Map<String, String> segmentToInstanceMap = selectionResult.getSegmentToInstanceMap();
+    Map<ServerInstance, List<String>> serverInstanceToSegmentsMap =
+        getServerInstanceToSegmentsMap(tableNameWithType, selectionResult.getSegmentToInstanceMap());
+    Map<ServerInstance, List<String>> serverInstanceToOptionalSegmentsMap =
+        getServerInstanceToSegmentsMap(tableNameWithType, selectionResult.getOptionalSegmentToInstanceMap());
+    return new RoutingTable(merge(serverInstanceToSegmentsMap, serverInstanceToOptionalSegmentsMap),
+        selectionResult.getUnavailableSegments(), selectionResult.getNumPrunedSegments());
```

Comment:
Can we merge these 3 methods into one (i.e. process mandatory and optional together)?
IMO that has 2 benefits:
- Logic more clear
- Reduce overhead, only need to maintain one map instead of 3

Reply:
sgtm